test/idle-source-e2e/idle_source_test.go

```diff
@@ -70,6 +86,58 @@ func (is *IdleSourceSuite) TestIdleKeyedReducePipeline() {
 	done <- struct{}{}
 }
 
+func (is *IdleSourceSuite) TestIdleKeyedReducePipelineWithKafkaSource() {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	topic := CreateKafkaTopic("2", "1")
+	fileData, err := os.ReadFile("testdata/kafka-pipeline.yaml")
+	is.NoError(err)
+	updatedFileData := strings.ReplaceAll(string(fileData), "my-topic", topic)
```

Comment:
@KeranYang  this line is required now to update the kafka topic in pipeline.yaml file

Reply:
It can be removed if we [define the pipeline object](https://github.com/numaproj/numaflow/blob/main/test/kafka-e2e/kafka_test.go#L43) instead of reading from a template. But it's fine we can keep it.